core/arch/arch.c

```diff
@@ -3545,6 +3550,15 @@ priv_mcontext_to_dr_mcontext(dr_mcontext_t *dst, priv_mcontext_t *src)
      */
     if (dst->size > sizeof(dr_mcontext_t))
         return false;
+#if defined(AARCH64)
+    /* Clients built before support of Arm AArch64's Scalable Vector Extension
+     * (SVE) are not binary compatible with the latest build.
```

Comment:
I would explain that we *could* support binary compatibility because of the user-set size field but have chosen not to put in the effort (otherwise future readers may be puzzled as to why it isn't using the size field to set the old layout, like it's doing on x86).